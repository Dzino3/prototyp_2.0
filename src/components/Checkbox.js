import React from "react";
import { useState } from "react";
import './style/showStatusStayle.css';

//Subcomponent for filter functions
const Checkbox = ({ label, checked, ...props }) => {
    const [isChecked, setIsChecked] = useState(checked);
  return (
    <div className="checkbox-wrapper">
      <label>
        <input className="checkBoxInput" type="checkbox" checked={isChecked} 
        onChange={() => setIsChecked((prev) => !prev)} {...props}></input>
        <span className="spanLabel">{label}</span>
      </label>
    </div>
  );
};
export default Checkbox;