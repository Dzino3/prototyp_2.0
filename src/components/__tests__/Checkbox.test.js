import React from "react";
import renderer from 'react-test-renderer';
import Checkbox from '../Checkbox';

import { useState as useStateMock } from "react";

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}))

describe(`Rendering Checkbox `, () => {
    const setState = jest.fn()

    beforeEach(() => {
      useStateMock.mockImplementation(init => [init, setState])
    })

    test('should render Checkbox', function () {
        let labelX = "Hallo42";
        let checkedX = "true";

        let tree = renderer.create(<Checkbox label={labelX} checked={checkedX}/>);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('checkbox state', () => {
      
      let tree = renderer.create(
        <Checkbox />
      )
      expect(setState).toHaveBeenCalledTimes(0)
      expect(tree).toBeTruthy()
     
    })
})