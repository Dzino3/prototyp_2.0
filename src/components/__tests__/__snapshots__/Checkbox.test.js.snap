// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Checkbox  should render Checkbox 1`] = `
<div
  className="checkbox-wrapper"
>
  <label>
    <input
      checked="true"
      className="checkBoxInput"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="spanLabel"
    >
      Hallo42
    </span>
  </label>
</div>
`;
